specs/data_structures.md

```diff
@@ -379,16 +380,21 @@ Finally, the `availableDataRoot` of the block [Header](#header) is computed as t
 
 A share is a fixed-size data chunk that will be erasure-coded and committed to in [Namespace Merkle trees](#namespace-merkle-tree).
 
-| name      | type               | description     |
-| --------- | ------------------ | --------------- |
-| `rawData` | `byte[SHARE_SIZE]` | Raw share data. |
+| name          | type                         | description                |
+| ------------- | ---------------------------- | -------------------------- |
+| `namespaceID` | [NamespaceID](#type-aliases) | Namespace ID of the share. |
+| `rawData`     | `byte[SHARE_SIZE]`           | Raw share data.            |
 
 An example layout of the share's internal bytes is shown below. For non-parity shares _with a reserved namespace_, the first `SHARE_RESERVED_BYTES` bytes (`*` in the figure) is the starting byte of the first request in the share as an unsigned integer, or `0` if there is none. In this example, the first byte would be `80` (or `0x50` in hex). For shares _with a non-reserved namespace_ (and parity shares), the first `SHARE_RESERVED_BYTES` bytes have no special meaning and are simply used to store data like all the other bytes in the share.
 
 ![fig: Reserved share.](./figures/share.svg)
 
 For non-parity shares, if there is insufficient request data to fill the share, the remaining bytes are padded with `0`.
 
+### Share Serialization
+
+Shares [canonically serialized](#serialization) using only the raw share data, i.e. `serialize(share) = serialize(share.rawData)`.
```

Comment:
Just for my understanding: In the context of the NMT this means, that a leaf (share) will get hashed in the following way:
`nsid||nsid||hash(leafPrefix||leaf)`, where `leaf =  raw_data` 
as opposed to:
`nsid||nsid||hash(leafPrefix||leaf)`, where `leaf = nid || raw_data`

https://github.com/liamsi/nmt-experiments/blob/70b02631b0e79986ec266b514e6da607e5c8e9d7/trillian_based/hasher.go#L48

Reply:
Right.